src/test/groovy/com/lesfurets/jenkins/TestInterceptingGCL.groovy

```diff
@@ -0,0 +1,243 @@
+                        .defaultVersion("gamma")
+                        .allowOverride(false)
+                        .implicit(false)
+                        .targetPath(sharedLibTwo)
+                        .retriever(projectSource(sharedLibTwo))
+                        .build()
+        helper.registerSharedLibrary(library)
+
+        runScript('job/library/cross_class/test_annotation.jenkins')
+        printCallStack()
+        assertCallStack().contains("""ClassA.methodA()""")
+        assertCallStack().contains("""ClassA.sh(echo 'ClassA: I'm field of A')""")
+        assertCallStack().contains("""ClassB.methodB()""")
+        assertCallStack().contains("""ClassB.sh(echo 'ClassB: I'm field of B')""")
+    }
+
+    @Test
+    void test_cross_lib_interop_library_loaded_with_implicity() throws Exception {
+        def libraryA = library().name('test_cross_class_uno')
+                        .defaultVersion("alpha")
+                        .allowOverride(false)
+                        .implicit(true)
+                        .targetPath(sharedLibTwo)
+                        .retriever(projectSource(sharedLibTwo))
+                        .build()
+        helper.registerSharedLibrary(libraryA)
+        def libraryB = library().name('test_cross_class_uno')
+                        .defaultVersion("alpha")
+                        .allowOverride(false)
+                        .implicit(true)
+                        .targetPath(sharedLibTwo)
+                        .retriever(projectSource(sharedLibTwo))
+                        .build()
+        helper.registerSharedLibrary(libraryA)
+
+        runScript('job/library/cross_class/test_implicity.jenkins')
+        printCallStack()
+        assertCallStack().contains("""ClassA.methodA()""")
+        assertCallStack().contains("""ClassA.sh(echo 'ClassA: I'm field of A')""")
+        assertCallStack().contains("""ClassB.methodB()""")
+        assertCallStack().contains("""ClassB.sh(echo 'ClassB: I'm field of B')""")
+    }
+
+    @Test
+    void test_cross_lib_interop_no_implicity_dynamic() throws Exception {
+        def library = library().name('test_cross_class_dos')
+                        .defaultVersion("beta")
+                        .allowOverride(false)
+                        .implicit(false)
+                        .targetPath(sharedLibTwo)
+                        .retriever(projectSource(sharedLibTwo))
+                        .build()
+        helper.registerSharedLibrary(library)
+
+        runScript('job/library/cross_class/test_dynamic.jenkins')
+        printCallStack()
+        assertCallStack().contains("""ClassA.methodA()""")
+        assertCallStack().contains("""ClassA.sh(echo 'ClassA: I'm field of A')""")
+        assertCallStack().contains("""ClassB.methodB()""")
+        assertCallStack().contains("""ClassB.sh(echo 'ClassB: I'm field of B')""")
+    }
+
+    @Test
+    void test_cross_lib_interop_no_implicity_annotation() throws Exception {
```

Comment:
Cross lib tests are incorrect. 
scenarios should be like:
1. load libA, libB
2. Test pipeline can call vars from the both libs
3. Test libB can call LibA vars
4. Test case if both LibA and LibB have a vars/method with the same signature (who wins?) 
4.1 Verify how it works in real jenins   

Test the same (if possible) with classes (what if you call a static Method). 
Verify the behaviour is the same in real Jenkins instance